packages/react-i18n/src/connection.ts

```diff
@@ -0,0 +1,34 @@
+import {TranslationDictionary} from './types';
+import {noop} from './utilities';
+
+export interface TranslationGetter {
+  (locale: string):
+    | TranslationDictionary
+    | Promise<TranslationDictionary | undefined>
```

Comment:
Tiny nit: can we reorder this so that the promise is last, so that it's a bit more obvious that this is `MaybePromise<TranslationDictionary | undefined>`